pglogical/hooks.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from odoo.sql_db import Cursor
+from odoo.tools import config
+
+SECTION_NAME = "pglogical"
+
+
+def post_load():
+    """
+    Patch cursor to funnel DDL through pglogical.replicate_ddl_command if configured to
+    do so
+    """
+
+    _logger = logging.getLogger("odoo.addons.pglogical")
+
+    if SECTION_NAME not in config.misc:
+        _logger.info("%s section missing in config, not doing anything", SECTION_NAME)
+        return
+    replication_sets = list(
+        filter(None, config.misc[SECTION_NAME].get("replication_sets", "").split(","))
+    )
+    if not replication_sets:
+        _logger.error("no replication sets defined, not doing anything")
+        return
+    if config["test_enable"]:
+        _logger.info("test mode enabled, not doing anything")
+        return
+
+    _logger.info("patching cursor to intercept ddl")
+    execute_org = Cursor.execute
+
+    def execute(self, query, params=None, log_exceptions=None):
+        """Wrap DDL in pglogical.replicate_ddl_command"""
+        if (query[:6] == "CREATE" or query[:5] == "ALTER" or query[:4] == "DROP") and (
+            # don't replicate constraints, triggers, indexes
+            "CONSTRAINT" not in query
+            and "TRIGGER" not in query
+            and "INDEX" not in query
```

Comment:
Here too, I am a little concerned about casing. However, when normalizing the rest of the query, there could be a false positive from table names like `text_index` or `update_trigger` so would you consider (if you have a match on the operation type above) to split the query by whitespace and check if you have a full word match on one of these terms rather than just a subset?

Reply:
I think at this point we can use sqlparse as @antonioburic proposed above, then we're sure